examples/cis2-wccd/src/lib.rs

```diff
@@ -1187,13 +1612,257 @@ mod tests {
+        let mut host = TestHost::new(state, state_builder);
+
+        // Call the contract function.
+        let result: ContractResult<()> = contract_set_paused(&ctx, &mut host);
+
+        // Check the result.
+        claim!(result.is_ok(), "Results in rejection");
+
+        // Check contract is paused.
+        claim_eq!(host.state().paused, true, "Smart contract should be paused");
+    }
+
+    /// Test unpausing the contract.
+    #[concordium_test]
+    fn test_unpause() {
+        // Set up the context.
+        let mut ctx = TestReceiveContext::empty();
+        ctx.set_sender(ADMIN_ADDRESS);
+
+        // Set up the parameter to pause the contract.
+        let parameter_bytes = to_bytes(&true);
+        ctx.set_parameter(&parameter_bytes);
+
+        // Set up the state and host.
+        let mut state_builder = TestStateBuilder::new();
+        let state = initial_state(&mut state_builder);
+        let mut host = TestHost::new(state, state_builder);
+
+        // Call the contract function.
+        let result: ContractResult<()> = contract_set_paused(&ctx, &mut host);
+
+        // Check the result.
+        claim!(result.is_ok(), "Results in rejection");
+
+        // Check contract is paused.
+        claim_eq!(host.state().paused, true, "Smart contract should be paused");
+
+        // Set up the parameter to unpause the contract.
+        let parameter_bytes = to_bytes(&false);
+        ctx.set_parameter(&parameter_bytes);
+
+        // Call the contract function.
+        let result: ContractResult<()> = contract_set_paused(&ctx, &mut host);
+
+        // Check the result.
+        claim!(result.is_ok(), "Results in rejection");
+
+        // Check contract is unpaused.
+        claim_eq!(host.state().paused, false, "Smart contract should be unpaused");
+    }
+
+    /// Test that only the current admin can pause/unpause the contract.
+    #[concordium_test]
+    fn test_pause_not_authorized() {
+        // Set up the context.
+        let mut ctx = TestReceiveContext::empty();
+        // NEW_ADMIN is not the current admin but tries to pause/unpause the contract.
+        ctx.set_sender(NEW_ADMIN_ADDRESS);
+
+        // Set up the parameter to pause the contract.
+        let parameter_bytes = to_bytes(&true);
+        ctx.set_parameter(&parameter_bytes);
+
+        // Set up the state and host.
+        let mut state_builder = TestStateBuilder::new();
+        let state = initial_state(&mut state_builder);
+        let mut host = TestHost::new(state, state_builder);
+
+        // Call the contract function.
+        let result: ContractResult<()> = contract_set_paused(&ctx, &mut host);
+
+        // Check that invoke failed.
+        claim_eq!(
+            result,
+            Err(ContractError::Unauthorized),
+            "Pause should fail because not the current admin tries to invoke it"
+        );
+    }
+
+    /// Test that one can NOT call state-mutative functions (wrap, unwrap,
```

Comment:
```suggestion
    /// Test that one can NOT call non-admin functions (wrap, unwrap,
```
Or whatever you decide to change it to in the previous location 😊